src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -111,35 +111,35 @@ internal Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
             var languageWorkerChannel = _languageWorkerChannelFactory.CreateLanguageWorkerChannel(_scriptOptions.RootScriptPath, _workerRuntime, _metricsLogger, attemptCount, _managedDependencyOptions);
             languageWorkerChannel.SetupFunctionInvocationBuffers(_functions);
             _jobHostLanguageWorkerChannelManager.AddChannel(languageWorkerChannel);
-            languageWorkerChannel.StartWorkerProcessAsync()
-                 .ContinueWith(workerInitTask =>
-                 {
-                     if (workerInitTask.IsCompleted)
-                     {
-                         _logger.LogDebug("Adding jobhost language worker channel for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
-                         languageWorkerChannel.SendFunctionLoadRequests();
-                         State = FunctionDispatcherState.Initialized;
-                     }
-                     else
-                     {
-                         _logger.LogWarning("Failed to start language worker process for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
-                     }
-                 });
+            _logger.LogDebug("Adding jobhost language worker channel for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
+            // Not awaiting
+            languageWorkerChannel.StartWorkerProcess();
+            languageWorkerChannel.SendFunctionLoadRequests().ContinueWith(functionLoadTask =>
+                {
+                    if (functionLoadTask.IsCompleted)
+                    {
+                        State = FunctionDispatcherState.Initialized;
+                    }
+                    else
+                    {
+                        _logger.LogWarning("Failed to start language worker process for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
+                    }
+                });
             return Task.CompletedTask;
         }
 
         internal async void InitializeWebhostLanguageWorkerChannel()
         {
             _logger.LogDebug("Creating new webhost language worker channel for runtime:{workerRuntime}.", _workerRuntime);
-            ILanguageWorkerChannel workerChannel = await _webHostLanguageWorkerChannelManager.InitializeChannelAsync(_workerRuntime);
+            ILanguageWorkerChannel workerChannel = _webHostLanguageWorkerChannelManager.CreateChannel(_workerRuntime);
```

Comment:
this needs to be async as it involves starting process

Reply:
Process.Start is a [sync method](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.start?view=netcore-2.2), and previously the only async code was waiting for the init code. 

Is the reason we want to keep this async so it's running on a separate thread?